framework_tvb/tvb/adapters/exporters/export_manager.py

```diff
@@ -124,19 +167,23 @@ def export_data(self, data, exporter_id, project):
         if not exporter.accepts(data):
             raise InvalidExportDataException("Current data can not be exported by specified exporter")
 
-        # Now compute and create folder where to store exported data
-        # This will imply to generate a folder which is unique for each export
-        data_export_folder = None
-        try:
-            data_export_folder = self._build_data_export_folder(data)
-
-            # Here is the real export                    
-            self.logger.debug("Start export of data: %s" % data.type)
-            export_data = exporter.export(data, data_export_folder, project)
-        finally:
-            # In case export did not generated any file delete folder
-            if data_export_folder is not None and len(os.listdir(data_export_folder)) == 0:
-                os.rmdir(data_export_folder)
+        dt_list = self.get_data_with_references_list(data)
+        if len(dt_list) > 1:
+            export_data = self.export_data_with_references(dt_list, project, exporter)
+        else:
+            # Now compute and create folder where to store exported data
+            # This will imply to generate a folder which is unique for each export
+            data_export_folder = None
+            try:
+                data_export_folder = self._build_data_export_folder(data)
+
+                # Here is the real export
+                self.logger.debug("Start export of data: %s" % data.type)
+                export_data = exporter.export(data, data_export_folder, project)
+            finally:
+                # In case export did not generated any file delete folder
```

Comment:
Maybe this part can be enhanced and both scenarios (single file or zip refences) have a try/finally?